src/Kiota.Builder/Writers/Shell/ShellCodeMethodWriter.cs

```diff
@@ -442,7 +465,15 @@ protected virtual void WriteCommandHandlerBody(CodeMethod codeElement, RequestPa
                 requestMethod = "SendNoContentAsync";
             }
 
-            writer.WriteLine($"{(isVoid ? string.Empty : "var response = ")}await RequestAdapter.{requestMethod}(requestInfo, errorMapping: {errorMappingVarName}, cancellationToken: {cancellationTokenParamName});");
+            var pageInfo = codeElement?.PagingInformation;
+            if (pageInfo != null)
+            {
+                writer.WriteLine($"var pagingData = new PageLinkData(requestInfo, Stream.Null, responseFormat: ResponseFormat.JSON, itemName: \"{pageInfo.ItemName}\", nextLinkName: \"{pageInfo.NextLinkName}\");");
```

Comment:
why not reading the response format from the response content type header instead?

Reply:
Because we only support JSON at the moment. Should I add another data format as well?

Reply:
What I'm struggling to understand is why does this need to be "hardcoded by code-gen" when this information in facts comes back with the response headers?

Reply:
Because there's no handler for any other format. I'll update to make it clearer.